### PR TITLE
New features for this week.

### PR DESCRIPTION
**Pull request by krockpot**
See code in doInBackground.

To test, delete the Fake Test sheet on our testing spreadsheet and change the app to write to that sheet (when it doesn't exist). The app will create the sheet and write to it (as usual).

I will remove the FAKE_TEST enum before merging.

### CODE REVIEW

**Comment by krockpot** [comment]
This resolves #10

File: sheets436/src/main/java/edu/umd/cmsc436/sheets/Sheets.java [hunk 1]
```diff
@@ -200,6 +200,7 @@ static void showGooglePlayErrorDialog(Host host, Activity hostActivity) {
         RH_POP("'Balloon Test (RH)'"),
         LH_CURL("'Curling Test (LH)'"),
         RH_CURL("'Curling Test (RH)'"),
+        FAKE("'Fake Test'"),
```
Comment (MrPickles) [inline sheets436/src/main/java/edu/umd/cmsc436/sheets/Sheets.java]: Don't have unless enums. Users are exposed to this. Run the actual enums on new spreadsheets.

File: sheets436/src/main/java/edu/umd/cmsc436/sheets/WriteDataTask.java [hunk 2]
```diff
@@ -133,6 +137,35 @@ protected Exception doInBackground(WriteData... params) {
...
+                    try {
+                        // Update the request
+                        sheetsService.spreadsheets().batchUpdate(spreadsheetId, batchReq).execute();
+                        // Retry our original write
+                        if (wd.central) {
+                            this.writeToCentral(wd);
+                        } else {
+                            this.writeToPrivate(wd);
+                        }
+                    } catch (Exception e2) {
```
Comment (MrPickles) [inline sheets436/src/main/java/edu/umd/cmsc436/sheets/WriteDataTask.java]: It looks kinda like callback hell, except with try-catch. Is there any way you can refactor to make it cleaner (e.g. make the try blocks tighter)? When I was thinking about having a small, single layer try-catch around each call to `execute`.

**Review by MrPickles — CHANGES_REQUESTED** [review]

**Comment by krockpot** [comment]
Resolves #5

File: sheets436/src/main/java/edu/umd/cmsc436/sheets/Sheets.java (hunk 1 above)
Comment (krockpot) [reply sheets436/src/main/java/edu/umd/cmsc436/sheets/Sheets.java]: Removed / tested on completely blank spreadsheet.

File: sheets436/src/main/java/edu/umd/cmsc436/sheets/WriteDataTask.java (hunk 2 above)
Comment (krockpot) [reply sheets436/src/main/java/edu/umd/cmsc436/sheets/WriteDataTask.java]: Separated into helper function.

**Comment by krockpot** [comment]
Changes complete. Please re-review @MrPickles .

**Review by MrPickles — APPROVED** [review]

**Comment by MrPickles** [comment]
LGTM

**Pull request merged**